src/server/src/modules/facultyMember.ts

```diff
@@ -20,32 +21,41 @@ export const createFacultyMember = (user: IUser) => {
  * @description updates an existing faculty member profile in the database
  * @param id number
  * @param user User
- * @param departmentId string
+ * @param department Department
  * @param websiteLink string
  * @param office string
  * @param title string
  * @returns Promise
  */
 export const updateFacultyMember = async (
     user: IFacultyMember['user'],
-    departmentId: IFacultyMember['departmentId'],
+    department: IFacultyMember['department'],
     websiteLink: IFacultyMember['websiteLink'],
     office: IFacultyMember['office'],
     title: IFacultyMember['title'],
     id: number
 ) => {
+    const departmentRepository = getRepository(Department);
+    const facultyMemberRepository = getRepository(FacultyMember);
 
-
-    const facultyToUpdate = await getRepository(FacultyMember).findOne(id);
+    const facultyToUpdate = await facultyMemberRepository.findOne(id);
     if (facultyToUpdate !== undefined) {
+        if (department !== undefined) {
+            const departmentObject = await departmentRepository.findOne(department.id)
+            if (departmentObject !== undefined) {
+                facultyToUpdate.department = departmentObject;
+                await facultyMemberRepository.save(facultyToUpdate);
+            }
+        }
+
         await getRepository(User).update(user.id, {
             biography: user.biography,
             firstName: user.firstName,
             middleName: user.middleName,
             lastName: user.lastName,
         });
-        return await getRepository(FacultyMember).update(id, {
-            departmentId,
+        return await facultyMemberRepository.update(id, {
+            department,
```

Comment:
I don't think we need to pass the department here

Reply:
Done.